src/plugins/telemetry/public/services/telemetry_sender.test.ts

```diff
@@ -127,67 +127,103 @@ describe('TelemetrySender', () => {
       expect(telemetryService.getIsOptedIn).toBeCalledTimes(0);
       expect(shouldSendReport).toBe(false);
     });
+  });
+  describe('sendIfDue', () => {
+    let originalFetch: typeof window['fetch'];
+    let mockFetch: jest.Mock<typeof window['fetch']>;
 
-    describe('sendIfDue', () => {
-      let originalFetch: typeof window['fetch'];
-      let mockFetch: jest.Mock<typeof window['fetch']>;
+    beforeAll(() => {
+      originalFetch = window.fetch;
+    });
 
-      beforeAll(() => {
-        originalFetch = window.fetch;
-      });
+    beforeEach(() => (window.fetch = mockFetch = jest.fn()));
+    afterAll(() => (window.fetch = originalFetch));
 
-      beforeEach(() => (window.fetch = mockFetch = jest.fn()));
-      afterAll(() => (window.fetch = originalFetch));
+    it('does not send if shouldSendReport returns false', async () => {
+      const telemetryService = mockTelemetryService();
+      const telemetrySender = new TelemetrySender(telemetryService);
+      telemetrySender['shouldSendReport'] = jest.fn().mockReturnValue(false);
+      telemetrySender['retryCount'] = 0;
+      await telemetrySender['sendIfDue']();
```

Comment:
(Thinking out loud, not related to the PR as it was already present) This is a code smell of bad isolation. Proper unit testing should never require to mock the blackboxed component.

e.g `shouldSendReport` mocking could be replaced with `telemetryService.canSendTelemetry` mocking (which would be the correct way to test the method's behavior btw). Here, we need to mock some of the `TelemetrySender`'s methods to test the behavior of other methods of the class, which is **not** unit testing the class.